### PR TITLE
Add a supervision endpoint to delete all nodes builds in a study

### PR DESCRIPTION
**Pull request by sBouzols**

### CODE REVIEW

**Review by jamal-khey — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/SupervisionController.java [hunk 1]
```diff
@@ -95,4 +95,13 @@ public ResponseEntity<Long> getIndexedTombstonedEquipmentsCount() {
     public ResponseEntity<Long> deleteStudyIndexedEquipmentsAndTombstoned(@PathVariable("studyUuid") UUID studyUuid) {
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(supervisionService.deleteStudyIndexedEquipmentsAndTombstoned(studyUuid));
     }
+
+    @DeleteMapping(value = "/studies/{studyUuid}/nodes/builds")
+    @Operation(summary = "Invalidate node builds for the given study")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "all built nodes for the given study have been invalidated")})
+    public ResponseEntity<Void> invalidateAllNodeBuilds(@PathVariable("studyUuid") UUID studyUuid) {
+        supervisionService.invalidateAllNodeBuilds(studyUuid);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/SupervisionController.java:103]: Invoke directly the `StudyService`

File: src/main/java/org/gridsuite/study/server/service/SupervisionService.java [hunk 2]
```diff
@@ -197,5 +197,9 @@ private Map<UUID, String> formatSubreportMap(String subReporterKey, List<Network
             node -> node.getId() + "@" + subReporterKey)
         );
     }
+
+    public void invalidateAllNodeBuilds(UUID studyUuid) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/SupervisionService.java:201]: Remove this method

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 3]
```diff
@@ -1450,6 +1449,12 @@ public void moveStudySubtree(UUID studyUuid, UUID parentNodeToMoveUuid, UUID ref
         notificationService.emitElementUpdated(studyUuid, userId);
     }
 
+    @Transactional
+    public void invalidateAllNodeBuilds(UUID studyUuid) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:1453]: invalidateAllNode**s**Builds ?

Comment (sBouzols) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java]: fixed

File: src/main/java/org/gridsuite/study/server/service/SupervisionService.java (hunk 2 above)
Comment (sBouzols) [reply src/main/java/org/gridsuite/study/server/service/SupervisionService.java]: Kept and removed from `StudyService` as discussed

File: src/main/java/org/gridsuite/study/server/SupervisionController.java (hunk 1 above)
Comment (sBouzols) [reply src/main/java/org/gridsuite/study/server/SupervisionController.java]: Will add logs in `SupervisionService::invalidateAllNodeBuilds` and pass `StudyService::invalidateBuild` public

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**